gist.el

```diff
@@ -186,7 +194,7 @@ With a prefix argument, makes a private paste."
 
 (defun gist-created-callback (gist)
   (let ((location (oref gist :html-url)))
-    (gist-list-reload t)
+    (gist-list-reload "" t)
```

Comment:
nil and "" aren't synonyms for the current user. nil means the current gist buffer (not necessarily belonging to the current user), whereas "" means the current user. The complication is that gist-list-reload needs to handle both of these cases robustly since it can be interactively invoked from any gist-list buffer, through M-x gist-list-reload, or by a user modifying their own gists.

The problem with passing a username instead of "" to gist-list-reload from gist-created-callback is that we have to add code to gist-list-user just to determine whether the username passed is equivalent to the current user, i.e.

```elisp
(defun gist-list-user (username &optional force-reload background)
  (interactive
   (let ((username (read-from-minibuffer "GitHub user: " nil nil nil
                                          'gist-user-history))
         (force-reload (equal current-prefix-arg '(4))))
     (list username force-reload)))
  ;; if buffer exists, it contains the current gh profile
  (let* ((gh-profile-current-profile (or gh-profile-current-profile
                                         (gh-profile-completing-read)))
         (api (gist-get-api nil))
         (api-user (gh-api-get-username api))
         (username-arg-is-current-user (string= username api-user))
         (bufname (if (null username)
                      (if (not (equal major-mode 'gist-list-mode))
                          (error "Current buffer isn't a gist-list-mode buffer")
                        (buffer-name))
                    (format "*%s:%sgists*"
                            gh-profile-current-profile
                            (if (or (string= "" username)
                                    username-arg-is-current-user)
                                ""
                              (format "%s's-" username)))))
         (username (or (and (null username) gist-list-buffer-user)
                       (and (not (or (null username)
                                     (string= "" username)
                                     username-arg-is-current-user))
                            username)
                       api-user)))
; etc.
```